core/src/main/java/org/bitcoinj/core/NetworkParameters.java

```diff
@@ -176,17 +177,17 @@ public static NetworkParameters fromID(String id) {
     }
 
     /**
-     * Return network parameters for a {@link BitcoinNetwork} enum
+     * Return network parameters for a {@link Network}.
+     * <p>
+     * Alternative networks will be found if they have been registered with {@link Networks} registry.
      * @param network the network
      * @return the network parameters for the given string ID
      * @throws IllegalArgumentException if unknown network
      */
     public static NetworkParameters of(Network network) {
-        if (network instanceof BitcoinNetwork) {
-            return BitcoinNetworkParams.of((BitcoinNetwork) network);
-        } else {
-                throw new IllegalArgumentException("Unknown network");
-        }
+        return (network instanceof BitcoinNetwork)
+                ? BitcoinNetworkParams.of((BitcoinNetwork) network)
+                : Networks.find(network).orElseThrow(() -> new IllegalArgumentException("Unknown network"));
```

Comment:
Is this change strictly needed in this commit, or could it be separated?

Reply:
It is required for this commit to work properly. (If an address is created from an alt-`Network`, we need to be able to retrieve the corresponding alt-`NetworkParameters`.)  However, it could be placed in a separate commit.